src/wp-includes/class-wp-block.php

```diff
@@ -140,6 +140,33 @@ public function __construct( $block, $available_context = array(), $registry = n
 
 		$this->available_context = $available_context;
 
+		$this->refresh_context_dependents();
+		$this->refresh_parsed_block_dependents();
+	}
+
+	/**
+	 * Updates the context for the current block and its inner blocks.
+	 *
+	 * This method updates the block's `context` property by merging the available context from ancestor blocks
+	 * with the block's own context values. It removes specific context keys (`postId` and `postType`) that should
+	 * not be carried over. The block only consumes the context keys specified in its registered block type (`uses_context`).
+	 *
+	 * After updating its own context, the method also updates the context of inner blocks, if any, by passing down
+	 * any context values the block provides (`provides_context`).
+	 *
+	 * If the block has inner blocks, the method recursively processes them by creating new instances of `WP_Block`
+	 * for each inner block and updating their context based on the block's `provides_context` property.
+	 *
+	 * @since 6.7.0
+	 */
+	public function refresh_context_dependents() {
+		$this->context = $this->available_context;
+
+		// Remove "postId" and "postType" keys.
+		if ( isset( $this->context['postId'] ) || isset( $this->context['postType'] ) ) {
```

Comment:
What is the reasoning for this special handling that didn't exist before?

Reply:
It's definitely an issue with unit tests, but not the code. I don't expect we would ever have to explicitly clean the context with the harcocded list of properties like here.

Reply:
What is better way to tackle these?